packages/cli-kit/src/public/node/context/local.ts

```diff
@@ -99,7 +99,7 @@ export function alwaysLogAnalytics(env = process.env): boolean {
  * @returns True if SHOPIFY_CLI_1P is truthy.
  */
 export function firstPartyDev(env = process.env): boolean {
-  return isTruthy(env[environmentVariables.firstPartyDev])
+  return isSpin(env) || isTruthy(env[environmentVariables.firstPartyDev])
```

Comment:
This will return true if you are running the CLI inside the Spin instance. But I guess you also want it to be true if you are running the CLI locally and connecting to a remote spin instance no?

```suggestion
  return isSpin(env) || isSpinEnvironment(env) || isTruthy(env[environmentVariables.firstPartyDev])
```